### PR TITLE
drivers: sensor: fxos8700: Update driver to use DEVICE_DT_GET

### PR DESCRIPTION
**Pull request by benjaminbjornsson**

### CODE REVIEW

**Review by gmarull — DISMISSED** [review]

**Review by gmarull — DISMISSED** [review]

File: drivers/sensor/fxos8700/fxos8700.c [hunk 1]
```diff
@@ -387,33 +387,31 @@ static int fxos8700_init(const struct device *dev)
...
 	struct sensor_value odr = {.val1 = 6, .val2 = 250000};
-	const struct device *rst;
 
 	if (!device_is_ready(config->i2c.bus)) {
 		LOG_ERR("I2C bus device not ready");
 		return -ENODEV;
 	}
 
-	if (config->reset_name) {
+	if (config->reset_gpio.port->name) {
```
Comment (MaureenHelm) [inline drivers/sensor/fxos8700/fxos8700.c:396]:
```suggestion
	if (config->reset_gpio.port) 
```

File: drivers/sensor/fxos8700/fxos8700_trigger.c [hunk 2]
```diff
@@ -16,13 +16,13 @@ static void fxos8700_gpio_callback(const struct device *dev,
 {
 	struct fxos8700_data *data =
 		CONTAINER_OF(cb, struct fxos8700_data, gpio_cb);
+	const struct fxos8700_config *config = dev->config;
```
Comment (MaureenHelm) [inline drivers/sensor/fxos8700/fxos8700_trigger.c:19]: `dev` is a GPIO device here, not a sensor device

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

File: drivers/sensor/fxos8700/fxos8700.c (hunk 1 above)
Comment (benjaminbjornsson) [reply drivers/sensor/fxos8700/fxos8700.c]: Done!

File: drivers/sensor/fxos8700/fxos8700_trigger.c (hunk 2 above)
Comment (benjaminbjornsson) [reply drivers/sensor/fxos8700/fxos8700_trigger.c]: Fixed!

**Review by MaureenHelm — APPROVED** [review]

File: drivers/sensor/fxos8700/fxos8700_trigger.c (hunk 2 above)
Comment (MaureenHelm) [reply drivers/sensor/fxos8700/fxos8700_trigger.c]: Thank you!

Comment (benjaminbjornsson) [reply drivers/sensor/fxos8700/fxos8700_trigger.c]: Well thank you, appreciate your thorough reviews!

**Review by gmarull — APPROVED** [review]

**Pull request merged**